src/pokemon_special_anim_scene.c

```diff
@@ -1145,197 +1214,264 @@ static void sub_811E300(struct Sprite * sprite)
+    LoadCompressedSpriteSheet(&spriteSheet);
+    LoadSpritePalette(&spritePalette);
+    taskId = CreateTask(Task_LevelUpVerticalSprites, 0);
+    gTasks[taskId].tXpos = x - 32;
+    gTasks[taskId].tYpos = y + 32;
+    gTasks[taskId].tTileTag = tileTag;
+    gTasks[taskId].tPaletteTag = paletteTag;
+    gTasks[taskId].tPriority = priority;
+    gTasks[taskId].tSubpriority = subpriority;
     SetGpuReg(REG_OFFSET_BLDCNT, BLDCNT_EFFECT_NONE  | BLDCNT_TGT2_ALL);
     SetGpuReg(REG_OFFSET_BLDALPHA, BLDALPHA_BLEND(12, 6));
 }
 
-bool8 sub_811E680(void)
+bool8 LevelUpVerticalSpritesTaskIsRunning(void)
 {
-    return FuncIsActiveTask(sub_811E694);
+    return FuncIsActiveTask(Task_LevelUpVerticalSprites);
 }
 
-static void sub_811E694(u8 taskId)
+static void Task_LevelUpVerticalSprites(u8 taskId)
 {
     s16 *data = gTasks[taskId].data;
-    switch (data[0])
+    switch (tState)
     {
     case 0:
-        if (data[3] == 0)
+        if (tTimer == 0)
         {
-            data[3]++;
-            sub_811E710(taskId, data);
-            if (data[2] > 17)
-                data[0]++;
+            tTimer++;
+            CreateLevelUpVerticalSprite(taskId, data);
+            if (tMadeSprCt > 17)
+                tState++;
         }
         else
         {
-            data[3]++;
-            if (data[3] == 2)
-                data[3] = 0;
+            tTimer++;
+            if (tTimer == 2)
+                tTimer = 0;
         }
         break;
     case 1:
-        if (data[1] == 0)
+        if (tActiveSprCt == 0)
         {
-            FreeSpriteTilesByTag(data[6]);
-            FreeSpritePaletteByTag(data[7]);
+            FreeSpriteTilesByTag(tTileTag);
+            FreeSpritePaletteByTag(tPaletteTag);
             DestroyTask(taskId);
         }
         break;
     }
 }
 
-static void sub_811E710(u8 taskId, s16 *data)
+static void CreateLevelUpVerticalSprite(u8 taskId, s16 *data)
 {
     u8 spriteId;
-    struct SpriteTemplate template = gUnknown_8459ACC;
-    template.tileTag = data[6];
-    template.paletteTag = data[7];
-    data[2]++;
-    spriteId = CreateSprite(&template, ((data[2] * 219) & 0x3F) + data[4], data[5], data[9]);
+    struct SpriteTemplate template = sSpriteTemplate_LevelUpVertical;
+    template.tileTag = tTileTag;
+    template.paletteTag = tPaletteTag;
+    tMadeSprCt++;
+    spriteId = CreateSprite(&template, ((tMadeSprCt * 219) & 0x3F) + tXpos, tYpos, tSubpriority);
     if (spriteId != MAX_SPRITES)
     {
-        gSprites[spriteId].oam.priority = data[8];
-        gSprites[spriteId].data[1] = 0;
-        gSprites[spriteId].data[2] = ((data[2] * 1103515245 + 24691) & 0x3F) + 0x20;
-        gSprites[spriteId].data[7] = taskId;
-        data[1]++;
+        gSprites[spriteId].oam.priority = tPriority;
+        gSprites[spriteId].tsYsubpixel = 0;
+        // similar to the LCRNG in random.c, but seeding from data[2]
+        gSprites[spriteId].tsSpeed = ((tMadeSprCt * 1103515245 + 24691) & 0x3F) + 0x20;
```

Comment:
I meant, it looks pointless since it's not a recurrence. tMadeSprCt is simply a counter